### PR TITLE
added ascii check validation

### LINKED ISSUE #845: Feature request - Validation/Constraint - Rows (or table) only contains characters in ASCII characters
# Overview
Hello, 

Is it possible to add a validation function that flags whether or not a given Resource contains ASCII only characters? 

Example of valid and not valid data:

```
# Valid
A,B,C
1,2,3

# Not valid
A,B,C
1,µ,€
```

Example implementation:
```
# Basic implementation
my_checks = [frictionless.checks.isascii()]
frictionless.validate(data, checks=my_checks)

# Alternative implementation
# Flagging if any data is outside a list of allowed unicode codes or
# Flagging if any data is within a list of disallowed unicode codes. 
codes= range(1 to 100)
my_checks = [frictionless.checks.codes(whitelist=codes)]

codes= [85, 164, 302]
my_checks = [frictionless.checks.charset(blacklist=codes]
```

Thanks

---

Please preserve this line to notify @roll (lead of this repository)

### PR DESCRIPTION
**Pull request by shashigharti**
* Added check for ascii string/character validation
* Added new tests

- fixes #845
- Looking for feedback on:
```

def test_validate_ascii():
    report = validate(
        "data/ascii.csv",
        checks=[{"code": "ascii-value", "ForbiddenValues": [2]}],
    )
    assert report.flatten(["rowPosition", "fieldPosition", "code"]) == [
        [2, 2, "forbidden-value"],
    ]
```
This tests does not pass through because of this:
`def __init__(self, descriptor=None, *, forbidden_values=[]):`

forbidden_values is not set to None; if I set it to None other tests fail for this kind of scenario:
`checks=[checks.ascii_value()]`

### CODE REVIEW

File: frictionless/checks/cell/ascii_value.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+from ... import errors
+from ...check import Check
+from typing import Union, Iterator
+
+
+class ascii_value(Check):
+    """Check for forbidden values in a field
```
Comment (roll) [inline frictionless/checks/cell/ascii_value.py]: I think `check whether all string characters in the data are ASCII`

File: frictionless/checks/cell/ascii_value.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    Parameters:
+       forbidden_values? (any[]): a list of forbidden values
+
+    """
+
+    code = "ascii-value"
+    Errors = [errors.NotAscii]
+
+    def __init__(
+        self, descriptor: Union[list, None] = None, *, forbidden_values: list = []
```
Comment (roll) [inline frictionless/checks/cell/ascii_value.py]: I would remove `forbidden_values` option as we already have `checks.forbidden_value`

File: frictionless/checks/cell/ascii_value.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                note = "the value is not ascii"
+                yield errors.NotAscii.from_row(row, note=note, field_name=field_name)
+
+    # Metadata
+
+    metadata_profile = {
+        "type": "object",
+        "required": ["ForbiddenValues"],
+        "properties": {
+            "ForbiddenValues": {"type": "array"},
```
Comment (roll) [inline frictionless/checks/cell/ascii_value.py]: Let's remove `forbiddenValues` option (note if we had to keep it the first letter is lowercase)

File: frictionless/checks/cell/forbidden_value.py [hunk 4]
```diff
@@ -16,7 +16,7 @@ class forbidden_value(Check):
     Parameters:
        descriptor (dict): check's descriptor
        field_name (str): a field name to look into
-       forbidden (any[]): a list of forbidden values
+       values (any[]): a list of forbidden values
```
Comment (roll) [inline frictionless/checks/cell/forbidden_value.py:19]: :+1:

File: frictionless/checks/cell/ascii_value.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    ):
+        self.setinitial("ForbiddenValues", forbidden_values)
+        super().__init__(descriptor)
+        self.__forbidden_values = self.get("ForbiddenValues", [])
+
+    # Validate
+
+    def validate_row(self, row: any) -> Iterator[any]:
+        for field_name, cell in row.items():
+            not_ascii = False
```
Comment (roll) [inline frictionless/checks/cell/ascii_value.py]: Logic suggestion:

```python
for field in row.fields():
    if field.type == 'string':
        cell = row[field.name]
        if cell and not cell.isascii():
            note = "the cell contains non-ascii characters"
            yield errors.NotAscii.from_row(row, note=note, field_name=field.name)
```

File: frictionless/errors/cell.py [hunk 6]
```diff
@@ -143,3 +143,10 @@ class SequentialValueError(CellError):
     name = "Sequential Value"
     template = "The cell {cell} in row at position {rowPosition} and field {fieldName} at position {fieldPosition} has an error: {note}"
     description = "The value is not sequential."
+
+
+class NotAscii(CellError):
```
Comment (roll) [inline frictionless/errors/cell.py]: I would name it `AsciiValueError` similar to the existent errors

File: frictionless/errors/cell.py [hunk 7]
```diff
@@ -143,3 +143,10 @@ class SequentialValueError(CellError):
...
+    code = "not-ascii"
```
Comment (roll) [inline frictionless/errors/cell.py]: `non-ascii`

File: frictionless/errors/cell.py [hunk 8]
```diff
@@ -143,3 +143,10 @@ class SequentialValueError(CellError):
...
+    name = "Not Ascii Value"
+    template = "The cell {cell} in row at position {rowPosition} and field {fieldName} at position {fieldPosition} has an error: {note}"
+    description = "The value is not ascii."
```
Comment (roll) [inline frictionless/errors/cell.py]: The cell contains non-ascii characters

**Review by roll — CHANGES_REQUESTED** [review]
@shashigharti 
Looks good! I added a few suggestions

File: frictionless/checks/cell/ascii_value.py (hunk 5 above)
Comment (shashigharti) [reply frictionless/checks/cell/ascii_value.py]: @roll isascii() only works for python 3.7 and above; it doesn't work in 3.6. Due to that I used the above method. Shall I replace it with isascii()?

**Comment by shashigharti** [comment]
Thank you! @roll . I have made the changes and it is ready for review. One question that I have is:

"isascii() only works for python 3.7 and above; it doesn't work in 3.6"

**Comment by roll** [comment]
@shashigharti 
Let's skip tests for Python3.6 for this feature (we will drop Python3.6 soon in v5) - see in other tests how it's done. My guess that `str.isascii` is way more performant than the enconde version (you can try checking the speed on large synthetic tables on in cycles I might be wrong)

**Comment by shashigharti** [comment]
> @shashigharti Let's skip tests for Python3.6 for this feature (we will drop Python3.6 soon in v5) - see in other tests how it's done. My guess that `str.isascii` is way more performant than the enconde version (you can try checking the speed on large synthetic tables on in cycles I might be wrong)

Thank you!. Sure I will check and update the changes.

**Review by roll — CHANGES_REQUESTED** [review]
Great work @shashigharti!

All look good jut please add the docs to https://framework.frictionlessdata.io/docs/guides/validation-checks:
- for `checks.ascii_value`

And we also need to add docs for the new steps to https://framework.frictionlessdata.io/docs/guides/transform-steps:
- for `steps.field_merge`
- for `steps.field_pack`

I think we can use this PR for all of the mentioned additions

File: tests/checks/cell/test_ascii_value.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
+from frictionless import validate, checks
+
+
+# Issues
+
+
+def test_validate_ascii_value_845():
```
Comment (roll) [inline tests/checks/cell/test_ascii_value.py:9]: BTW don't forget that you need to skip these tests for Python3.6 (see other skipped tests for reference)

Comment (shashigharti) [reply tests/checks/cell/test_ascii_value.py:9]: sure!. Sorry I forgot to add it in last commit.

**Comment by shashigharti** [comment]
@roll  Thank you. I have made the requested changes and it is ready for the review.

**Review by roll — APPROVED** [review]
Great thanks!

**Pull request merged**